pkg/reconciler/v1alpha1/revision/revision.go

```diff
@@ -447,3 +455,43 @@ func (c *Reconciler) updateStatus(desired *v1alpha1.Revision) (*v1alpha1.Revisio
 	existing.Status = desired.Status
 	return c.ServingClientSet.ServingV1alpha1().Revisions(desired.Namespace).UpdateStatus(existing)
 }
+
+// TODO(643) Change this logic in 0.5 to only drop the deprecated label
+//           Delete this logic in 0.6
+func (c *Reconciler) migrateConfigurationMetadata(rev *v1alpha1.Revision) (*v1alpha1.Revision, error) {
+	stale := false
+
+	if rev.Annotations[serving.ConfigurationGenerationLabelKey] != "" {
```

Comment:
Can you document the steps of the logic here?  There are things that are clearly subtle (e.g. looking up a label key in an annotations map) that may be lost on folks.